internal/services/applications/application_password_resource.go

```diff
@@ -30,7 +25,74 @@ func applicationPasswordResource() *schema.Resource {
 			return err
 		}),
 
-		Schema: aadgraph.PasswordResourceSchema("application_object_id"),
+		Schema: map[string]*schema.Schema{
+			"application_object_id": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validate.UUID,
+			},
+
+			"key_id": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validate.UUID,
+			},
+
+			"display_name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"description"},
+				ForceNew:      true, // TODO: ForceNew not required with MS Graph, remove in v2.0
+			},
+
+			// TODO: remove in v2.0
+			"description": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"display_name"},
+				Deprecated:    "This property has been renamed to `display_name` and will be removed in version 2.0 of the AzureAD provider",
+				ForceNew:      true,
+			},
+
+			"value": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Sensitive:    true,
+				Deprecated:   "[NOTE] In version 2.0 of the AzureAD provider, this attribute will become read-only as it will no longer be possible to specify a password value. It will be generated by Azure Active Directory and persisted to state for reuse in your Terraform configuration.",
+				ValidateFunc: validation.StringLenBetween(1, 863), // Encrypted secret cannot be empty and can be at most 1024 bytes.
+			},
```

Comment:
we could conditionally change the behaviour of this field in the Beta to be either Required & ForceNew or Computed? e.g.

```suggestion
			"value": func() *schema.Schema {
				if msGraphEnabled {
					return &schema.Schema{
						Type:         schema.TypeString,
						Computed:     true,
						Sensitive:    true,
					}
				}
				
				return &schema.Schema{
					Type:         schema.TypeString,
					Required:     true,
					ForceNew:     true,
					Sensitive:    true,
					Deprecated:   "[NOTE] In version 2.0 of the AzureAD provider, this attribute will become read-only as it will no longer be possible to specify a password value. It will be generated by Azure Active Directory and persisted to state for reuse in your Terraform configuration.",
					ValidateFunc: validation.StringLenBetween(1, 863), // Encrypted secret cannot be empty and can be at most 1024 bytes.
				}
			}
			}(),
```

whilst not ideal it seems fine with a flag in the documentation during the beta?

Reply:
Agreed, this makes sense 👍

Reply:
I think this comes back to one of the initial poblems we had, in that the provider and resource schemas are declared before the provider is configured. Not sure if there's a way to defer this until after it's been determined whether the beta is enabled?

Reply:
@tombuildsstuff This has gotten really fun.

In addition to the `value` field, the `key_id` (`keyId`) field is also no longer allowed to be user-supplied. However, despite [documentation to the contrary](https://docs.microsoft.com/en-us/graph/api/application-addpassword?view=graph-rest-beta&tabs=http), the remaining fields `display_name`, `start_date` and `end_date` are completely ignored by the API. It assigns null, now and now+2yrs respectively for each of those.

So I've opted to reject those properties at apply time when MS Graph is used, with a view to making them all Computed in v2.0. I'll raise the issue of the other fields being ignored and then hopefully we can make them Optional in future. WDYT?